### PR TITLE
feat(argo-apps): migrate argocd-notifications to argo-apps

### PR DESCRIPTION
**Pull request by hairmare**
# Description

⚠️ unless you used argocd notifications (and you probably didn't) this is a no-op change

Create a new `argo-apps` and move Argo CD Notifications to it (from `infra-apps`). 

AFAICT no one has used it in infra apps so far, i also plan on adding some more argo things (like argocd-applicationsets and rollouts) to the new `argo-apps` down the line.

# Issues

# Checklist

* [x] I updated the version in Chart.yaml
* [x] I updated applicable README.md files using  `pre-commit run`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] If I updated a dependency tool, or app, this PR contains a short summary of the changes I'm pulling
* [x] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

File: charts/argo-apps/ci/default-values.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+argocdNotifications:
+  enabled: true
+  values:
+    TODO: add values in values.yaml
```
Comment (tongpu) [inline charts/argo-apps/ci/default-values.yaml:4]: What's the idea behind this `TODO` line?

**Review by tongpu — COMMENTED** [review]
This will create a new release for `infra-apps`, because of that I would suggest to explicitly mention at the top of the PR that this is an "empty" release of `infra-apps`.

Have we thought about how we could maintain changelogs for our charts in some way, because that would make it easier to track changes in GitHub Releases and maybe also ArtifactHub?

File: charts/argo-apps/ci/default-values.yaml (hunk 1 above)
Comment (hairmare) [reply charts/argo-apps/ci/default-values.yaml:4]: tbh, it's mostly cargo culted from elsewhere

**Comment by hairmare** [comment]
> suggest to explicitly mention at the top of the PR

@tongu I updated the PR and will also add a note in the commit during merge

**Review by tongpu — APPROVED** [review]

**Pull request merged**